File: MovieProject/MovieProject.Web/Modules/Northwind/SalesInvoice/SalesInvoiceColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace MovieProject.Northwind.Columns;

[ColumnsScript("Northwind.SalesInvoice")]
[BasedOnRow(typeof(SalesInvoiceRow), CheckNames = true)]
public class SalesInvoiceColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int SaleInvoiceId { get; set; }
    public DateTime Date { get; set; }
    [EditLink]
    public string Notes { get; set; }
}

File: MovieProject/MovieProject.Web/Modules/ServerTypes/Northwind/SalesInvoiceColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SalesInvoiceRow } from "./SalesInvoiceRow";

export interface SalesInvoiceColumns {
    SaleInvoiceId: Column<SalesInvoiceRow>;
    Date: Column<SalesInvoiceRow>;
    Notes: Column<SalesInvoiceRow>;
}

export class SalesInvoiceColumns extends ColumnsBase<SalesInvoiceRow> {
    static readonly columnsKey = 'Northwind.SalesInvoice';
    static readonly Fields = fieldsProxy<SalesInvoiceColumns>();
}